openml/runs/run.py

```diff
@@ -707,3 +707,46 @@ def _to_dict(self) -> dict[str, dict]:  # noqa: PLR0912, C901
                                 current,
                             )
         return description
+
+    def _check_file_size(self, file: Any, max_file_size_mb: int = 100) -> bool:
+        """Check file size for file-like objects"""
+        file_size = 0
+        if isinstance(file, (bytes, bytearray)):  # If file is a binary data
+            file_size = len(file)
+        elif hasattr(file, "read") and hasattr(file, "seek"):  # File-like object
+            current_pos = file.tell()  # Save the current position
+            file.seek(0, 2)  # Move to the end of the file
+            file_size = file.tell()
+            file.seek(current_pos)  # Restore the original position
+        else:
+            raise TypeError("Unsupported file type. Provide binary data or a file-like object.")
+
+        if file_size > max_file_size_mb * 1024 * 1024:  # MB in bytes
+            return False
+        return True
+
+    def add_file_to_run(self, file: Any, name: str, max_file_size_mb: int = 100) -> None:
```

Comment:
I agree with Lennart we should just have proper support if we add this, e.g., simply keep a `files: list[Path]` property and update the `_get_file_elements` function accordingly.